### PR TITLE
Dev/hew/week9

### PR DESCRIPTION
**Pull request by DarthThunder**
It all works, I'm not sure I did it exactly how you wanted it though.

### CODE REVIEW

File: Week6Threading/Week6Threading/MathOps.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Week6Threading
```
Comment (mattandccu) [inline Week6Threading/Week6Threading/MathOps.cs]: You have a bunch of changes in this branch that are unrelated to the week9 solution. In a production environment, it is important to ensure that commits to a branch only contain the changes you intend to include. Otherwise, your team may become confused about what is/is not intended and the repository may get cluttered with dead code. (-5)

Comment (DarthThunder) [inline Week6Threading/Week6Threading/MathOps.cs]: Oh, sorry about that. I accidentally made the branch off of my last week7 project instead of the master, that's probably why that happened.

Date: Tue, 15 Mar 2016 13:05:32 -0700

File: Week9Networking/Week9Networking/Program.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        /// <summary>
+        /// Entery point to the program.
+        /// </summary>
+        /// <param name="args">Paramater for args.</param>
+        static void Main(string[] args)
+        {
+            int number;
+            Console.WriteLine("Insert a port number...");
+            var portNumber = Console.ReadLine();
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/Program.cs]: I was kind of envisioning the port number would be passed in via the "args" parameter, but your solution works just fine as well. And, I like that you let the user try a port number and then just use a default.

File: Week9Networking/Week9Networking/Program.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+            bool isNumber = Int32.TryParse(portNumber, out number);
+
+            if(isNumber == true)
+            {
+                WebServer server = new WebServer(SendResponse, "http://localhost:" + portNumber + "/");
+                server.Run();
+                Console.WriteLine("A simple webserver. Press a key to quit.");
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/Program.cs]: It would be good to tell the user what port number is being used, especially if you use a default port number.

File: Week9Networking/Week9Networking/Program.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                Console.ReadKey();
+                server.Stop();
+
+            }
+
+            else
+            {
+                WebServer server = new WebServer(SendResponse, "http://localhost:4858/");
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/Program.cs]: I really like your factoring here, separating the web server logic AND the request/response handling. It would allow other developers to add new response handling without having to modify your web server logic. Nice! (+2)

File: Week9Networking/Week9Networking/Program.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            bool isXInt = Int32.TryParse(xValue, out xNumber);
+            bool isYInt = Int32.TryParse(yValue, out yNumber);
+
+            Console.WriteLine(isXInt.ToString() + isYInt.ToString());
+
+            // Checks to see if x and y are ints.
+            if (isXInt == true && isYInt == true)
+            {
+                xNumber = Int32.Parse(xValue);
+                yNumber = Int32.Parse(yValue);
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/Program.cs]: You have already parsed the numbers above via TryParse, so you don't need to parse them again here.

File: Week9Networking/Week9Networking/WebServer.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+namespace Week9Networking
+{
+    /// <summary>
+    /// Creates a web server for a variety of uses.
+    /// </summary>
+    public class WebServer
+    {
+        private readonly HttpListener listener = new HttpListener();
+        private readonly Func<HttpListenerRequest, string> respondermethod;
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/WebServer.cs]: Nice use of a Func<> delegate! (+2)

File: Week9Networking/Week9Networking/WebServer.cs [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        /// <summary>
+        /// Sets up the web server, starts the listener.
+        /// </summary>
+        /// <param name="prefixes">Necessary prefixes for the server.</param>
+        /// <param name="method">Method for the response handling.</param>
+        public WebServer(string[] prefixes, Func<HttpListenerRequest, string> method)
+        {
+            if (!HttpListener.IsSupported)
+                throw new NotSupportedException(
+                    "Needs Windows XP SP2, Server 2003 or later.");
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/WebServer.cs]: Nice environment check here! (+1)

File: Week9Networking/Week9NetworkingTest/NetworkingTest.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Week9NetworkingTest
+{
+    [TestClass]
+    public class NetworkingTest
+    {
+        [TestMethod]
+        public void TestWebServerAddition()
+        {
+            WebServer server = new WebServer(Program.SendResponse, "http://localhost:4858/");
```
Comment (mattandccu) [inline Week9Networking/Week9NetworkingTest/NetworkingTest.cs]: Nice self-contained unit test! You even start the web server automatically.

File: Week9Networking/Week9Networking/WebServer.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        public void Run()
+        {
+            ThreadPool.QueueUserWorkItem((o) =>
+            {
+                Console.WriteLine("Webserver running...");
+                try
+                {
+                    while (listener.IsListening)
+                    {
+                        ThreadPool.QueueUserWorkItem((c) =>
```
Comment (mattandccu) [inline Week9Networking/Week9Networking/WebServer.cs]: Interesting approach that allows handling each incoming request on a different background thread.

**Comment by mattandccu** [comment]
Very nicely done, Heather! Way to work through the challenges to find a workable solution! Really like some of the factoring you did to ensure a good separation of concerns and increased flexibility.
